src/main/java/seedu/uninurse/storage/JsonAdaptedTask.java

```diff
@@ -31,6 +35,12 @@ public JsonAdaptedTask(@JsonProperty("taskDescription") String taskDescription,
     public JsonAdaptedTask(Task source) {
         taskDescription = source.getTaskDescription();
         dateTime = new JsonAdaptedDateTime(source.getDateTime());
+        if (source instanceof RecurringTask) {
+            RecurringTask recurTask = (RecurringTask) source;
+            freq = recurTask.recurrence;
+        } else {
+            freq = null;
+        }
```

Comment:
Similar to the suggestion above to avoid using `instanceof` by having the interface support a method that returns an `Optional<Frequency>`.